v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx

```diff
@@ -10,34 +10,58 @@ section: how to guides
 category: pallet design
 ---
 
-This guide will show you how to add the [Contracts pallet](/rustdocs/latest/pallet_contracts/index.html) to your runtime so that your blockchain can support Wasm smart contracts.
+This guide will show you how to add the [Contracts pallet](/rustdocs/latest/pallet_contracts/index.html) to your runtime so that your blockchain can support WebAssembly (Wasm) smart contracts.
+
 You can follow similar patterns to add additional FRAME pallets to your runtime, however you should note that each pallet is a little different in terms of the specific configuration settings needed to use it correctly.
 
+## What is the Contracts Pallet for?
+
+Broadly speaking there are two big use-cases you can apply it to.
+We'll only shortly describe those, for more information please check
+[the pallet reamde][pallet-readme].
+
+### Smart contracts as first-class citizens
+
+You want to build a general-purpose smart contracts chain, with them being the main focus of your chain.
+Maybe you have an innovative idea and want to have a smart contracts base layer which you complement
+with your chains unique selling point.
+
+### Smart contracts for users to customize how they interact with your chain
+
+You've built a chain that doesn't have smart contracts as first-class citizens,
+instead most of the logic is built using Substrate pallets that use weight-metering.
+
+But now you want to expose limited parts of your chains business logic to be customizable
+by users. Say you have e.g. a decentralized exchange chain and want to enable users
+uploading their own trading algorithms.
+Smart contracts are ideal for that ‒ they treat all user input as untrusted and potentially adversarial.
+With gas fees users have to pay for the execution time of their trading algorithms.
+
+The Contracts pallet provides the [Chain extension][chain-extension] primitive for exactly that.
+This way programming languages (like [ink!][ink]) can easily make use of your chains
+exposed business logic primitives.
+
 ## Before you begin
 
 Make sure you have the latest version of the Substrate node template compiled on your computer.
 If you haven't already done so, see [Create your first substrate chain](/tutorials/v3/create-your-first-substrate-chain/).
 
 ## Import the dependencies
 
-1. Add Contracts to your runtime.
-   To learn how to include Contracts in your runtime, see [Basic pallet integration](/how-to-guides/v3/basics/pallet-integration).
-
-1. Update `runtime/Cargo.toml` with the following entries:
+1. Add the following dependencies to your `runtime/Cargo.toml`:
    - `pallet-contracts`
    - `pallet-contracts-primitives`
+1. Integrate the Contracts pallet into your runtime as described in the
```

Comment:
The two RPC crates are missing here.

Reply:
They are added in a special section below (not my idea, the original how-to had this structure).